src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -2611,10 +2611,21 @@ void executeNodeActions()
 
             void executeNodeActionsByKind(ArrayBuilder<SyntaxNode> nodesToAnalyze, GroupedAnalyzerActions groupedActions, bool arePerSymbolActions)
             {
+                var analyzersForNodes = PooledHashSet<DiagnosticAnalyzer>.GetInstance();
```

Comment:
can this be lifted higher?

Reply:
yup, good idea!

Reply:
Actually not. It depends on groupdActions which differs from the different callers.

Reply:
but you can clear the results between the calls.  basically don't return to the pool just to fetch it out :)    the code isn't concurrent, so you can use the same scratch buffer multiple times, right?

Reply:
The pooledhashset doesn't show up at all in the allocations, so I'm inclined to keep this as is unless you feel strongly as this is a bit simpler.

Reply:
wfm